sympy/physics/continuum_mechanics/beam.py

```diff
@@ -974,8 +974,52 @@ def solve_for_reaction_loads(self, *reactions):
             eqs = deflection_curve.subs(x, position) - value
             deflection_eqs.append(eqs)
 
-        solution = list((linsolve([shear_curve, moment_curve] + shear_force_eqs + bending_moment_eqs + slope_eqs
-                            + deflection_eqs, (C3, C4) + reactions + rotation_jumps + deflection_jumps).args)[0])
+        equations = ([shear_curve, moment_curve]            +
+                 shear_force_eqs + bending_moment_eqs   +
+                 slope_eqs      + deflection_eqs)
+
+        unknowns  = (C3, C4) + reactions + rotation_jumps + deflection_jumps
+
+        try:
+            sol_tuple = linsolve(equations, unknowns).args[0]
+
+        except Exception as err:
+            support_syms = {load[0] for load in self._support_as_loads}
```

Comment:
Catching Exception will catch too many unexpected exception types. When using `try/except` you should be specific about what kinds of exception to catch.